api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -27,7 +27,15 @@ def on_connect(event, context)
   authorizer = request_context['authorizer']
 
   authorization_error_response = AuthErrorResponseHelper.get_response(authorizer)
-  return authorization_error_response unless authorization_error_response == nil
+  # If there is an authorization error, allow the websocket connection to open,
+  # but early return so we don't actually invoke the Build and Run lambda and try
+  # to create an SQS queue. We will send the auth error message via the websocket
+  # once the connection has been successfully made (in on_default).
+  if authorization_error_response
```

Comment:
can you add an additional comment that this is a pretty rare scenario because the auth error should be caught by the http authorizer?